src/command_modules/azure-cli-iot/azure/cli/command_modules/iot/custom.py

```diff
@@ -39,59 +38,123 @@ def iot_hub_create(client, name, resource_group_name,
+    result = client.create_or_update(resource_group_name=resource_group_name, resource_name=hub_name,
                                      iot_hub_description=iot_hub_description)
     return result
 
 
-def iot_hub_get_connection_string(client, name, resource_group_name, policy_name='iothubowner'):
-    access_policies = client.list_keys(resource_group_name, name).next()
+def iot_hub_list(client, resource_group_name=None):
+    if resource_group_name is None:
+        return client.list_by_subscription().next()
+    else:
+        return client.list_by_resource_group(resource_group_name).next()
+
+
+def iot_hub_show_connection_string(client, hub_name=None, resource_group_name=None, policy_name='iothubowner'):
+    if hub_name is None:
+        hubs = iot_hub_list(client, resource_group_name)
+        if hubs is None:
+            raise CLIError('No IoT Hub found.')
+        connection_strings = []
+        for h in hubs:
+            connection_string = get_single_iot_hub_connection_string(client, h.name, h.resourcegroup, policy_name)
+            connection_strings.append({"name": h.name,
+                                       "connectionString": connection_string})
+        return connection_strings
+    else:
+        if resource_group_name is None:
+            resource_group_name = get_resource_group_by_iot_hub_name(client, hub_name)
+        connection_string = get_single_iot_hub_connection_string(client, hub_name, resource_group_name, policy_name)
+        return {"connectionString": connection_string}
+
+
+def get_single_iot_hub_connection_string(client, hub_name, resource_group_name, policy_name):
+    access_policies = client.list_keys(resource_group_name, hub_name).next()
     if access_policies is None:
-        raise CLIError('No policy found from IoT Hub: {}.'.format(name))
+        raise CLIError('No policy found from IoT Hub: {}.'.format(hub_name))
     logger.info('Shared Access Polices: %s', str(access_policies))
     try:
         access_policy = next(x for x in access_policies
                              if policy_name.lower() == x.key_name.lower())
     except StopIteration:
-        raise CLIError('No policy found with name {}'.format(policy_name))
+        raise CLIError('No policy found with name {} from IoT Hub {}'.format(policy_name, hub_name))
 
-    connection_string = 'HostName={}.azure-devices.net;SharedAccessKeyName={};SharedAccessKey={}'.format(name,
-                                                                                                         policy_name,
-                                                                                                         access_policy.primary_key)
-    return {'connectionString': connection_string}
+    connection_string_template = 'HostName={}.azure-devices.net;SharedAccessKeyName={};SharedAccessKey={}'
+    return connection_string_template.format(hub_name, policy_name, access_policy.primary_key)
```

Comment:
Our pattern is to aim to return objects rather than just plain strings.
So this would `return {'connectionString': ...}`. This allows JSON format to print a valid JSON object.
See `az storage account show-connection-string` - https://github.com/Azure/azure-cli/blob/master/src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/custom.py#L67